### PR TITLE
feat: migrate to app router

### PR DESCRIPTION
**Pull request by VariableVic**
In this PR:
- Migrated all pages to App Router, with page, layout and loading.
- Moved collections fetching logic to edge api route
- RM /pages directory
- [WIP] Product module support - only locally for now

### CODE REVIEW

File: src/app/collections/[handle]/page.tsx [hunk 1]
```diff
@@ -0,0 +1,27 @@
+import CollectionTemplate from "@modules/collections/templates"
+import { Metadata } from "next"
+
+type Props = {
+  params: { handle: string }
+}
+
+const BASEURL = process.env.NEXT_PUBLIC_BASE_URL ?? "https://localhost:3000"
```
Comment (fPolic) [inline src/app/collections/[handle]/page.tsx:8]:
```suggestion
const BASEURL = process.env.NEXT_PUBLIC_BASE_URL ?? "http://localhost:3000"
```

File: src/app/collections/[handle]/page.tsx [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+  ).then((res) => res.json())
+
+  return {
+    title: `${collection.title} | Acme Store`,
+    description: `${collection.title} collection`,
+  }
+}
+
+export default async function CollectionPage({ params }: Props) {
+  const { collection } = await fetch(
```
Comment (fPolic) [inline src/app/collections/[handle]/page.tsx:22]: I'm getting an exception when I open collection details page

**Review by fPolic — COMMENTED** [review]
Great work with this one @VariableVic! LGTM overall 🚀 
Would like to hear what others think about the route handlers approach

**Review by kasperkristensen — APPROVED** [review]
LGTM! Only thing I would try to improve before merging is the `medusaRequest`. I think it would be great if we could improve a bit on handling query params, so people don't have to manually specify things as an array like this:

```
return await medusaRequest("GET", "/collections", {
    query: { "handle[]": handle },
  })
 ```
 
 And instead, if the value is an array then we add the `[]` to the key.

**Comment by VariableVic** [comment]
> LGTM! Only thing I would try to improve before merging is the `medusaRequest`. I think it would be great if we could improve a bit on handling query params, so people don't have to manually specify things as an array like this:
> 
> ```
> return await medusaRequest("GET", "/collections", {
>     query: { "handle[]": handle },
>   })
> ```
> 
> And instead, if the value is an array then we add the `[]` to the key.

Good suggestion @kasperkristensen, I'll fix that today

**Review by srindom — CHANGES_REQUESTED** [review]
Nice work just a few smaller adjustments afaict :)

File: src/app/collections/route.ts [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+      products,
+      count,
+    },
+    nextPage,
+  })
+}
+
+async function fetchCollection(handle: string) {
+  return await medusaRequest("GET", "/collections", {
+    query: { "handle[]": handle },
```
[the remark this answers is not in the document]
Comment (VariableVic) [reply src/app/collections/route.ts]: @srindom so this isn't valid?

```
query: { handle: ["one", "two"] } // -> handle[]=one,two
```

File: src/app/collections/[handle]/page.tsx (hunk 2 above)
Comment (VariableVic) [reply src/app/collections/[handle]/page.tsx]: @fPolic @srindom i can't reproduce this, are you still getting the error after I changed the fallback to http? 
Can you try to add these .env vars?

NEXT_PUBLIC_MEDUSA_BACKEND_URL=http://localhost:9000
NEXT_PUBLIC_BASE_URL=http://localhost:3000

Comment (VariableVic) [reply src/app/collections/[handle]/page.tsx]: Fixed thanks to Frane, the starter runs on port 8000 by default, not 3000

**Pull request merged**